src/Controls/src/Core/Handlers/Items/CarouselViewHandler.Windows.cs

```diff
@@ -521,15 +522,19 @@ void OnCollectionItemsSourceChanged(object sender, NotifyCollectionChangedEventA
 			SetCarouselViewPosition(carouselPosition);
 		}
 
-		void InitialSetup(object sender, SizeChangedEventArgs e)
-		{
-			if (e.NewSize.Width > 0 && e.NewSize.Height > 0)
-			{
-				ListViewBase.SizeChanged -= InitialSetup;
+		void OnListViewSizeChanged(object sender, SizeChangedEventArgs e) => Resize(e.NewSize);
 
-				if (_scrollViewer != null)
-					_scrollViewer.SizeChanged -= InitialSetup;
+		void OnScrollViewSizeChanged(object sender, SizeChangedEventArgs e)
+		{
+			//If there's a scroll viewer, it's enough to resize based on its size changed event, so we can avoid two event handlers doing the same
+			ListViewBase.SizeChanged -= OnListViewSizeChanged;
+			Resize(e.NewSize);
```

Comment:
Maybe check that the newSize or that the eventArgs aren't null? This might be the cause of the NRE I saw